xio-core/src/main/java/com/xjeffrose/xio/http/Client.java

```diff
@@ -78,11 +82,26 @@ public ChannelFuture write(Request request) {
       ChannelPromise promise = channel.newPromise();
       PromiseCombiner combiner = new PromiseCombiner();
       combiner.add(future.addListener(connectionListener));
-      combiner.add(channel.writeAndFlush(request).addListener(writeListener));
+      if (request.endOfMessage()) {
+        combiner.add(channel.writeAndFlush(request).addListener(writeListener));
+      } else {
+        combiner.add(channel.write(request).addListener(writeListener));
+      }
       combiner.finish(promise);
       return promise;
     } else {
-      return channel.writeAndFlush(request).addListener(writeListener);
+      if (request.endOfMessage()) {
+        return channel.writeAndFlush(request).addListener(writeListener);
+      } else {
+        return channel.write(request).addListener(writeListener);
+      }
+    }
+  }
+
+  public void prepareForReuse(Supplier<ChannelHandler> handlerSupplier) {
+    clientChannelInitializer.setAppHandler(handlerSupplier);
```

Comment:
FYI, If we're reusing an open connection the channel initializer isn't going to be invoked.